src/partition/repart.c

```diff
@@ -3682,11 +3682,15 @@ static int do_copy_files(
                 gid_t override_gid,
                 const Set *denylist) {
 
+        CopyFlags flags = COPY_REFLINK|COPY_HOLES|COPY_MERGE|COPY_REPLACE|COPY_SIGINT|COPY_HARDLINKS|COPY_ALL_XATTRS;
         int r;
 
         assert(p);
         assert(root);
 
+        if (streq(p->format, "vfat"))
+                flags |= COPY_SYMLINKS_GRACEFUL;
```

Comment:
Quite frankly, I'd just set this flag unconditionally, and not condition it on "vfat". If we add a log message about this to the output (as suggested earlier) this sounds OK to do.